socs/agents/meinberg_syncbox/agent.py

```diff
@@ -0,0 +1,513 @@
+            mbgSyncboxN2XPtpNwProt::
+                Options:: unknown(0),
+                          ipv4(1),
+                          ipv6(2),
+                          ieee802-3(3),
+                          deviceNet(4),
+                          controlNet(5),
+                          profiNet(6)
+            mbgSyncboxN2XPtpPortState::
+                Options:: uninitialized(0),
+                          initializing(1),
+                          faulty(2),
+                          disabled(3),
+                          listening(4),
+                          preMaster(5),
+                          master(6),
+                          passive(7),
+                          uncalibrated(8),
+                          slave(9)
+            mbgSyncboxN2XPtpDelayMechanism::
+                Options:: e2e(0),
+                          p2p(1)
+            mbgSyncboxN2XPtpTimescale::
+                Options:: tai(0),
+                          arb(1)
+            mbgSyncboxN2XPtpGrandmasterTimesource::
+                Options:: atomicClock(16),
+                          gps(32),
+                          terrestrialRadio(48),
+                          ptp(64),
+                          ntp(80),
+                          handSet(96),
+                          other(144),
+                          internalOscillator(160)
+            mbgSyncboxN2XPtpGrandmasterClockAccuracy::
+                Options:: accurateToWithin25ns(32),
+                          accurateToWithin100ns(33),
+                          accurateToWithin250ns(34),
+                          accurateToWithin1us(35),
+                          accurateToWithin2Point5us(36),
+                          accurateToWithin10us(37),
+                          accurateToWithin25us(38),
+                          accurateToWithin100us(39),
+                          accurateToWithin250us(40),
+                          accurateToWithin1ms(41),
+                          accurateToWithin2Point5ms(42),
+                          accurateToWithin10ms(43),
+                          accurateToWithin25ms(44),
+                          accurateToWithin100ms(45),
+                          accurateToWithin250ms(46),
+                          accurateToWithin1s(47),
+                          accurateToWithin10s(48),
+                          accurateToGreaterThan10s(49)
+        """
+
+        session.set_status('running')
+        self.is_streaming = True
+        while self.is_streaming:
+            yield dsleep(1)
+            if not self.connected:
+                self.log.error('No SNMP response. Check your connection!')
+                self.log.info('Trying to reconnect.')
+
+            read_time = time.time()
+
+            # Check if sample period has passed before getting status
+            if (read_time - self.lastGet) < self.sample_period:
+                continue
+
+            main_get_list = []
+            get_list = []
```

Comment:
This is confusing.... why two lists?  It looks like all things (or almost all things?) are "get"ted twice.